### PR TITLE
pin CPython 3.8 installer on macosx10.9

### PR DESCRIPTION
**Pull request by mayeut**
macosx11 installer requires macOS 11+ for CPython 3.8 while macosx11 installer for CPython 3.9+ only requires macOS 10.9+ on x86_64.

c.f. comments in #663

### CODE REVIEW

File: bin/update_pythons.py [hunk 1]
```diff
@@ -196,13 +196,19 @@ def __init__(self) -> None:
...
+    def update_version_macos(
+        self, identifier: str, version: Version, spec: Specifier
+    ) -> ConfigMacOS | None:
         sorted_versions = sorted(v for v in self.versions_dict if spec.contains(v))
 
-        for version in reversed(sorted_versions):
+        if version <= Version("3.8.9999"):
+            file_idents = ("macosx10.9.pkg",)
+        else:
+            file_idents = ("macos11.pkg", "macosx10.9.pkg")
```
Comment (mayeut) [inline bin/update_pythons.py:207]: Should we just drop the tuple here and just go ahead and define `file_ident` once and for all ?

Comment (mayeut) [reply bin/update_pythons.py]: Nevermind, I went this way just to silence the MyPy issue - I guess I could have solved differently.

**Review by joerick — APPROVED** [review]
looks good to me but I'll defer to @henryiii as he wrote this script originally

File: bin/update_pythons.py [hunk 2]
```diff
@@ -196,27 +196,32 @@ def __init__(self) -> None:
...
+    def update_version_macos(
+        self, identifier: str, version: Version, spec: Specifier
+    ) -> ConfigMacOS | None:
         sorted_versions = sorted(v for v in self.versions_dict if spec.contains(v))
 
-        for version in reversed(sorted_versions):
+        if version <= Version("3.8.9999"):
+            file_ident = "macosx10.9.pkg"
+        else:
+            file_ident = "macos11.pkg"
```
Comment (henryiii) [inline bin/update_pythons.py:207]: We could instead take the old file identifier, and only look for that.

Comment (henryiii) [reply bin/update_pythons.py:207]: This is fine for now, but that's a possible update later to remove the explicit version check.

**Review by henryiii — APPROVED** [review]

**Pull request merged**